### PR TITLE
[NET 6 Attribute Conversion] AddressBook

### PR DESCRIPTION
**Pull request by chamons**
Done with `./scripts/convert.sh AddressBook`.

No manual corrections and no fixes outside required using.

This part is strange:

```csharp
 	[SupportedOSPlatform ("maccatalyst14.0")]
	[SupportedOSPlatform ("maccatalyst14.0")]
	[UnsupportedOSPlatform ("maccatalyst14.0")]
	[UnsupportedOSPlatform ("maccatalyst14.0")]
```

However, it literally mirrors the legacy attributes that are doubled up. Thoguhts?

### CODE REVIEW

File: src/AddressBook/ABMultiValue.cs [hunk 1]
```diff
@@ -47,19 +47,19 @@
...
-#else
-	[UnsupportedOSPlatform ("ios9.0")]
-	[UnsupportedOSPlatform ("maccatalyst14.0")]
+#if NET
 	[SupportedOSPlatform ("maccatalyst14.0")]
+	[UnsupportedOSPlatform ("maccatalyst14.0")]
+	[UnsupportedOSPlatform ("ios9.0")]
 #if __MACCATALYST__
-	[Obsolete ("Starting with maccatalyst14.0 use the 'Contacts' API instead.", DiagnosticId = "BI1234", UrlFormat = "https://github.com/xamarin/xamarin-macios/wiki/Obsolete")]
+	[Obsolete ("Starting with maccatalyst14.0 Use the 'Contacts' API instead.", DiagnosticId = "BI1234", UrlFormat = "https://github.com/xamarin/xamarin-macios/wiki/Obsolete")]
```
Comment (rolfbjarne) [inline src/AddressBook/ABMultiValue.cs:55]: The capitalization here is not correct English.

**Review by rolfbjarne — APPROVED** [review]

File: src/AddressBook/ABMultiValue.cs (hunk 1 above)
Comment (chamons) [reply src/AddressBook/ABMultiValue.cs:55]: Ok, I'll teach mellite to lower case when gluing them together.

**Comment by rolfbjarne** [comment]
> This part is strange:
> 
> ```cs
>  	[SupportedOSPlatform ("maccatalyst14.0")]
> 	[SupportedOSPlatform ("maccatalyst14.0")]
> 	[UnsupportedOSPlatform ("maccatalyst14.0")]
> 	[UnsupportedOSPlatform ("maccatalyst14.0")]
> ```
> 
> However, it literally mirrors the legacy attributes that are doubled up. Thoguhts?

This is expected for some API, Apple added some API to Mac Catalyst and immediately deprecated them (iirc usually api that was already deprecated in iOS).

File: src/AddressBook/ABMultiValue.cs (hunk 1 above)
Comment (chamons) [reply src/AddressBook/ABMultiValue.cs]: Corrected in this and other active PRs - https://github.com/chamons/mellite/commit/1c4b860045fb3e3575948e65cdd5236c67f62fc3

**Review by dalexsoto — APPROVED** [review]

File: src/AddressBook/ABMultiValue.cs [hunk 2]
```diff
@@ -47,19 +47,19 @@
...
-	[UnsupportedOSPlatform ("maccatalyst14.0")]
+#if NET
 	[SupportedOSPlatform ("maccatalyst14.0")]
+	[UnsupportedOSPlatform ("maccatalyst14.0")]
+	[UnsupportedOSPlatform ("ios9.0")]
 #if __MACCATALYST__
 	[Obsolete ("Starting with maccatalyst14.0 use the 'Contacts' API instead.", DiagnosticId = "BI1234", UrlFormat = "https://github.com/xamarin/xamarin-macios/wiki/Obsolete")]
 #elif IOS
-	[Obsolete ("Starting with ios9.0 use the 'Contacts' API instead.", DiagnosticId = "BI1234", UrlFormat = "https://github.com/xamarin/xamarin-macios/wiki/Obsolete")]
+	[Obsolete ("Starting with ios9.0", DiagnosticId = "BI1234", UrlFormat = "https://github.com/xamarin/xamarin-macios/wiki/Obsolete")]
```
Comment (tj-devel709) [inline src/AddressBook/ABMultiValue.cs:57]: Do we include this obsolete if there is no error message?

File: src/AddressBook/ABPerson.cs [hunk 3]
```diff
@@ -713,60 +713,84 @@ public void RemoveImage ()
...
-#if !NET
-		[iOS (7,0)]
-#else
-		[UnsupportedOSPlatform ("ios7.0")]
-#if IOS
-		[Obsolete ("Starting with ios7.0 use 'GetCompositeNameFormat (null)' instead.", DiagnosticId = "BI1234", UrlFormat = "https://github.com/xamarin/xamarin-macios/wiki/Obsolete")]
+#if NET
+		[SupportedOSPlatform ("ios7.0")]
+		[SupportedOSPlatform ("maccatalyst14.0")]
+		[UnsupportedOSPlatform ("maccatalyst")]
```
Comment (tj-devel709) [inline src/AddressBook/ABPerson.cs:735]: Unsupported for maccatalyst but supported in maccatalyst 14.0 seems even more suspicous?

**Review by tj-devel709 — APPROVED** [review]
Few questions, but everything else looks good to me

**Comment by chamons** [comment]
This test seems to be failing relevant tests: AttributeTypoTest/Unavailable/Duplicates.

Looking into this.

**Comment by chamons** [comment]
This will fail until https://github.com/xamarin/xamarin-macios/pull/13745 lands and is merged. Fixing the '.' issue shortly.

**Comment by chamons** [comment]
Something screwy is going on in ABPerson.cs. Investigating...

**Review by rolfbjarne — APPROVED** [review]

**Review by tj-devel709 — APPROVED** [review]

**Comment by chamons** [comment]
Hmm, +.NET (BuildFailure) on Xtro. That's worth a rekick.

**Comment by chamons** [comment]
It is a valid issue, looking locally at it.

**Comment by chamons** [comment]
As mentioned in Teams, I am uncertain exactly why https://github.com/xamarin/xamarin-macios/pull/13715/commits/1176533cf0e609f980f4b148ebd0b011c10f8ead is needed but it seems sane.

**Comment by chamons** [comment]
I have no concerns about watchOS failures on this change, watch just being itself.

**Pull request merged**